popout.js

```diff
@@ -1,241 +1,355 @@
-						},
-						set: () => true
-					};
-                    canvas = new Proxy(() => canvas, handlers);
-                    Playlist.prototype.playSound = () => {}
-                    WebRTC.prototype.connect = () => {}
-				</script>`))
-		// Avoid having the UI initialized which renders the chatlog and all sorts
-		// of other things behind the sheet
-		body.append($(`<script>
-		      Game.prototype.initializeUI = function() {
-							ui.nav = new SceneNavigation();
-							ui.controls = new SceneControls();
-							ui.notifications = new Notifications().render();
-							ui.sidebar = new Sidebar();
-							// Back to initializeUI 
-							ui.players = new PlayerList();
-							ui.hotbar = new Hotbar();
-							ui.webrtc = new CameraViews(this.webrtc);
-							ui.pause = new Pause();
-							ui.menu = new MainMenu();
-
-							// sidebar elements only get created on the render
-							// but we don't want to render them
-							ui.chat = new ChatLog({tabName: "chat"})
-							ui.combat = new CombatTracker({tabName: "combat"})
-							ui.scenes = new SceneDirectory({tabName: "scenes"})
-							ui.actors = new ActorDirectory({tabName: "actors"})
-							ui.items = new ItemDirectory({tabName: "items"})
-							ui.journal = new JournalDirectory({tabName: "journal"})
-							ui.tables = new RollTableDirectory({tabName: "tables"})
-							ui.playlists = new PlaylistDirectory({tabName: "playlists"})
-							ui.compendium = new CompendiumDirectory({tabName: "compendium"})
-							ui.settings = new Settings({tabName: "settings"})
-					}
-							
-				KeyboardManager.prototype._onEscape = function(event, up, modifiers) {
-					if ( up || modifiers.hasFocus ) return;
-
-					// Case 1 - dismiss an open context menu
-					if ( ui.context && ui.context.menu.length ) ui.context.close();
-
-					// Case 2 - close open UI windows
-					else if ( Object.keys(ui.windows).length ) {
-						Object.values(ui.windows).filter(w => w.id !== ${sheetGetter}.id).forEach(app => app.close());
-					}
-
-					// Flag the keydown workflow as handled
-					this._handled.add(event.keyCode);
-				}
-				// Add delay before rendering in case some things aren't done initializing, like sheet templates
-				// which get loaded asynchronously.
-				Hooks.on('ready', () => setTimeout(() => PopoutModule.${renderer}(${sheetGetter}), ${timeout}));
-		  	window.dispatchEvent(new Event('load'))
-		  </script>`))
-		// Open new window and write the new html document into it
-		// We need to open it to the same url because some images use relative paths
-		let windowFeatures = undefined;
-		if (game.settings.get("popout", "useWindows")) {
-            const innerWidth = sheet.element.innerWidth() + padding * 2;
-            const innerHeight = sheet.element.innerHeight() + padding * 2;
-            const position = sheet.element.position();
+    constructor() {
+        this.poppedOut = new Map();
+        this.TIMEOUT_INTERVAL = 25 // ms
+        this.MAX_TIMEOUT = 250; // ms

```

Comment:
If someone's machine is struggling because it's checking the equality of a variable every 25ms, I would be surprised if it could run foundry at all :wink:  . But I have updated the timeout interval to 50ms anyway.

Reply:
hehe, no, not that it would be struggling because of the 25ms, but rather it could be struggling to render the DOM of the new sheet. Even on my powerful PC, Foundry will sometimes lag before a window opens up (depends on RAM usage, swapping, other chrome tabs open, disk speed and multiple other factors), so it was more about the 1 second timeout than about the 50ms interval. The point being, if the popout opens within 25ms or 50ms, it won't even be noticeable by the user, and by increasing the initial check interval, you avoid having "the first 7 checks will almost always fail" situation.